### ISSUE TITLE
Strange behavour with new method of download !

### ISSUE THREAD

**Issue by Nokia808** [jingyu9575/multithreaded-download-manager#138]
Hi. I started to suffer from this issue from 1st time I started to use new method when you 1st introduced it, but I did not report it because I was not on firefox 74. Now I received firefox 74 & still existing. 

Issue is that: for each download link, I have to try 1st trial then block it then retry again so as to achieve good speed of download ! Otherwise, if I try to download from a link for 1st trial & continue this trial, then speed download will be slow then increased gradually over time till reaching the max. speed (which some time achieved when reaching to 45% of size of file !).

I mean by max download speed: the speed of built-in firefox down-loader

Moreover, I mean by 1st & 2nd trial of download: per individual firefox session !

**Comment by jingyu9575** [comment]
> speed download will be slow then increased gradually over time till reaching the max

I'm not sure if this is due to the initial connection time. Is the initial connection slow on your network? The speed is calculated with the time since start, so e.g. if the initial connection takes 10 seconds and the download takes 10 seconds, the shown speed will be 1/2 of the actual speed.

**Comment by Nokia808** [comment]
@jingyu9575 
Yes ! I think yes. There was slow connection time. But I notice that when I received firefox 74, the slow connection time become too less frequent ! Before firefox 74 there were many download links showed slow connection time before starting downloads. But now on version 74 of firefox, such slow initial connection time become less frequent. 

But the question is that: why, on firefox bellow 74, when I close download window of your program then retried download link again, no slow connection time ??

Please keep this issue open so as to make me sure about it.

**Comment by Nokia808** [comment]
@jingyu9575 
Hi dear. After careful monitoring all the passed time, I can say with confidence the following:
EVEN with excellent Internet speed, for unknown reason, your add-on (some times not all times) suffer from slow initial connection & to overcome this user need to stop download & retry (re-click download link). At 2nd trail, there will be no any slow initial connection !!
